django/contrib/postgres/operations.py

```diff
@@ -33,11 +40,19 @@ def database_forwards(self, app_label, schema_editor, from_state, to_state):
     def database_backwards(self, app_label, schema_editor, from_state, to_state):
         if not router.allow_migrate(schema_editor.connection.alias, app_label):
             return
-        schema_editor.execute("DROP EXTENSION %s" % schema_editor.quote_name(self.name))
+
+        if self.extension_present(schema_editor, self.name):
+            schema_editor.execute("DROP EXTENSION %s" % schema_editor.quote_name(self.name))
+
         # Clear cached, stale oids.
         get_hstore_oids.cache_clear()
         get_citext_oids.cache_clear()
 
+    def extension_present(self, schema_editor, extension):
+        cur = schema_editor.connection.cursor()
+        cur.execute("SELECT * FROM pg_extension WHERE extname = %s", [extension])
```

Comment:
```python
with schema_editor.connection.cursor() as cursor:
    cursor.execute(...
```

Reply:
what is the advantage of using a context manager when i need to return `bool(cursor.fetchone())`?

Reply:
https://docs.djangoproject.com/en/3.0/topics/db/sql/#connections-and-cursors